File: Lecture9-Dilation/main.cpp
#include <opencv4/opencv2/opencv.hpp>

using namespace cv;

void dilation()
{
    Mat img = imread("/home/berkan/CLionProjects/OpenCV/Ders9_Dilation/word.jpg", IMREAD_GRAYSCALE);
    namedWindow("Dilation", WINDOW_AUTOSIZE);
    threshold(img, img, 128, 255, THRESH_BINARY);

    Mat filter, dilationImg;
    filter = Mat::ones(Size(3, 3), img.type());

    int min = 0;
    createTrackbar("Dilation", "Dilation", &min, 5);

    while (true)
    {
        dilate(img, dilationImg, filter, Point(-1, -1), min, 0, Scalar(255, 255, 255));
        imshow("Dilation", dilationImg);
        if (waitKey(10) == 27)
        {
            break;
        }
    }
    destroyAllWindows();
}

int main() {
    dilation();
    return 0;
}

File: Lecture9-Dilation/main.py
import cv2
import numpy as np

def nothing():
    pass

def dilation():
    img = cv2.imread("cumle.jpg", cv2.IMREAD_GRAYSCALE)
    cv2.namedWindow("Dilation Image", cv2.WINDOW_AUTOSIZE)
    thresh = cv2.threshold(img, 128, 255, cv2.THRESH_BINARY)
    filtre = np.ones((3,3))

    cv2.createTrackbar("Dilation", "Dilation Image", 0, 15, nothing)

    while True:
        dilationValue = cv2.getTrackbarPos("Dilation", "Dilation Image")
        dilationImg = cv2.dilate(img, filtre, iterations=dilationValue)
        cv2.imshow("Dilation Image", dilationImg)

        if cv2.waitKey(10) == 27:
            break

    cv2.destroyAllWindows()

if __name__ == '__main__':
    dilation()